File: PATTERNS/Pattern45/Pattern45.js
const readline = require('readline');

const r1  = readline.createInterface(process.stdin, process.stdout);

r1.setPrompt('Enter the number of rows');
r1.prompt();
r1.on('line' , (num) => {
    // upward pattern 
    for(let i =0;i<num;i++){
        let star = 2*i +1;
        let space = num-i;
        for(let j = 0;j < space;j++){
           process.stdout.write(' ');
        }
        for(let k =0;k<star;k++) {
            process.stdout.write('*');
        }
        
        console.log();
    }
    
    // downward pattern 
    for(let i =num-2 ;i>=0;i--){
        let star = 2*i +1;
        let space = num-i;
        for(let j = 0;j < space;j++){
           process.stdout.write(' ');
        }
        for(let k =0;k<star;k++) {
            process.stdout.write('*');
        }
        
        console.log();
    }
    
    r1.close();
})

File: PATTERNS/Pattern45/Pattern45.go
package main

import "fmt"

func main() {
	var n int
	var Corep int
	var spaceChange int
	fmt.Print("Enter the row number:")
	fmt.Scanf("%d",&n)
	for row:= 1; row <= 2*n-1; row++{
		if row<=n{
			if row > 1{
				Corep = 2*row-1
			}else{
				Corep = row
			}
		}else{
			Corep = 2*(2*n-row)-1
		}
		if row <= n{
			spaceChange = n-row+1
		}else{
			spaceChange = (row-n)+1
		}
		for space:= 1; space <= spaceChange; space++{
			fmt.Print(" ")
		}
		for col:= 1; col <= Corep; col++{
			fmt.Print("*");
		}
		fmt.Println()
	}
}
